src/pages/auth/verify/email/[token].page.tsx

```diff
@@ -9,34 +9,18 @@ import Confirmation from "@/components/extensive/Confirmation/Confirmation";
 import { IconNames } from "@/components/extensive/Icon/Icon";
 import BackgroundLayout from "@/components/generic/Layout/BackgroundLayout";
 import ContentLayout from "@/components/generic/Layout/ContentLayout";
-import { useVerificationUser } from "@/connections/VerificationUser";
-import { useOnMount } from "@/hooks/useOnMount";
+import { loadVerificationUser } from "@/connections/VerificationUser";
 import { useValueChanged } from "@/hooks/useValueChanged";
-import Log from "@/utils/log";
 
 const VerifyEmail = () => {
   const t = useT();
   const router = useRouter();
   const token = router.query.token as string;
   const [verified, setVerified] = useState(false);
 
-  const [, { requestFailed, isSuccess, sendVerifyUser }] = useVerificationUser({ token: token });
-
-  useValueChanged(requestFailed, () => {
-    if (requestFailed != null) {
-      Log.error("Failed to verify auth");
-      router.push("/");
-    }
-  });
-
-  useValueChanged(isSuccess, () => {
-    if (isSuccess) {
-      setVerified(true);
-    }
-  });
-
-  useOnMount(async () => {
-    sendVerifyUser();
+  useValueChanged(token, async () => {
+    const { isSuccess } = await loadVerificationUser({ token });
+    setVerified(isSuccess!);
```

Comment:
Please don't use connection loaders in components. The hooks are much more efficient. The `useState` for verified is also not needed, and neither is this `useValueChanged` hook. This should be sufficient:
```
const [, { isSuccess: verified }] = useVerificationUser({ token });
```

Reply:
The `useValueChanged()` for `fetchFailed` should likely be brought back though.